### PR TITLE
fix: replace wallet manage api

### PR DESCRIPTION
**Pull request by dudu0506**
# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/helpers/account.ts [hunk 1]
```diff
@@ -219,9 +219,12 @@ async function removeAccount(account: Account, signal?: AbortSignal) {
     }
 }
 
-export async function removeCurrentAccount(source: SocialSource) {
+export async function removeAccountByProfileId(source: SocialSource, profileId?: string) {
     const { accounts, currentProfile } = getProfileState(source);
-    const account = accounts.find((x) => isSameProfile(x.profile, currentProfile));
+    const account = accounts.find((x) => {
+        if (profileId) return x.profile.profileId === profileId;
```
Comment (guanbinrui) [inline src/helpers/account.ts:225]: It's much safer using `isSameProfile`.

File: src/app/(settings)/components/DisconnectButton.tsx [hunk 2]
```diff
@@ -1,38 +1,105 @@
...
-    if (!profiles.length) return [];
-    return Promise.all(
-        profiles.map((profile) => {
-            return getProfileById(narrowToSocialSource(profile.source), profile.identity);
-        }),
-    );
+const resolveFireflyProfiles = memoize(
+    async function (platforms: ProfileTab[]) {
+        if (!platforms.length) return [];
+        const lensIds = platforms.filter((x) => x.source === Source.Lens).map((x) => x.identity!);
```
Comment (guanbinrui) [inline src/app/(settings)/components/DisconnectButton.tsx:30]: `x.identity` could be `undefined`.

File: src/app/(settings)/components/DisconnectButton.tsx [hunk 3]
```diff
@@ -1,38 +1,105 @@
...
+        const farcasterIds = platforms.filter((x) => x.source === Source.Farcaster).map((x) => x.identity!);
```
Comment (guanbinrui) [inline src/app/(settings)/components/DisconnectButton.tsx:31]: Same here.

File: src/decorators/SetQueryDataForDeleteWallet.ts [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
         },
         (old) => {
             if (!old) return old;
             return produce(old, (draft) => {
-                for (const address of addresses) {
-                    draft.connected = draft.connected.filter((x) => x.profile.identity !== address);
-                    draft.related = draft.related.filter((x) => x.profile.identity !== address);
-                }
+                draft.connected = draft.connected.filter((x) => x.walletConnection.address !== address);
+                draft.related = draft.related.filter((x) => x.walletConnection.address !== address);
```
Comment (guanbinrui) [inline src/decorators/SetQueryDataForDeleteWallet.ts:32]: isSameAddress

File: src/app/(settings)/components/DisconnectButton.tsx [hunk 5]
```diff
@@ -1,38 +1,105 @@
...
+        case FireflyPlatform.Wallet:
+            return { source: FireflyPlatform.Wallet, identity: connection.address };
+        default:
+            safeUnreachable(connection.source);
+            return null;
+    }
+}
+
+async function updateAccountConnection(platform: FireflyPlatform, identity: string) {
+    const source = resolveSourceFromUrl(platform);
```
Comment (guanbinrui) [inline src/app/(settings)/components/DisconnectButton.tsx:60]: resolveSocialSourceInURL

File: src/helpers/account.ts (hunk 1 above)
Comment (dudu0506) [reply src/helpers/account.ts:225]: we only have profileId here, no real profile. And we compare with both source and profileId, same as isSameProfile

File: src/app/(settings)/components/DisconnectButton.tsx (hunk 5 above)
Comment (dudu0506) [reply src/app/(settings)/components/DisconnectButton.tsx:60]: resolveSocialSourceInURL means: **SocialSource** -> **SocialSourceInURL**. But here we need: **SourceInURL** -> **Source**

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**